src/util/logging.py

```diff
@@ -17,30 +17,47 @@
 
 from util.context import Context
 
+LOG_THROTTLING_LIMIT_PER_CALLER = 10
+
+log_call_count = dict()
+
+
+def log_multiline_message(message: Text, caller: Text):
+
+    # display logs called from one spot no more than the specified amount of times
+    if log_call_count.get(caller, 0) < LOG_THROTTLING_LIMIT_PER_CALLER:
+        log_call_count[caller] = log_call_count.get(caller, 0) + 1
+    elif log_call_count.get(caller, 0) == LOG_THROTTLING_LIMIT_PER_CALLER:
+        log_call_count[caller] = log_call_count.get(caller, 0) + 1
+        print(f"Logging calls from caller '{caller}' exceeded the throttling limit of",
+              f"{LOG_THROTTLING_LIMIT_PER_CALLER}. Further logs from this caller will be discarded")
+        return
+    else:
+        return
```

Comment:
This is a bit verbose would be a bit unreadable if e.g. I entered here just to see how messages are logged, I would have to read more into it than I would like to. I can give you an example of what I think might be better.

Btw of missing things because it's too verbose - is there a print missing in the first if, or did I just get lost?
Ok, I see it's there later - but I still got lost because of the verbosity of the first check. I'd extract the whole of those if to a separate method. It could be done directly (I recommend PyCharm Extract method if you don't know it), or here's my draft take on it, just to show you what I have in mind:
```
if check_log_limit_exceeded:
	return
```	

```	
def check_log_limit_exceeded(caller):
	log_limit_left = log_call_count.get(caller, 0) - LOG_THROTTLING_LIMIT_PER_CALLER
	if log_limit_left == 1:
           print(f"Logging calls from caller '{caller}' exceeded the throttling limit of",
              f"{LOG_THROTTLING_LIMIT_PER_CALLER}. Further logs from this caller will be discarded")
	bool is_loggable = log_limit_left == 0
	return is_loggable
```

Reply:
Done, thank you